src/main/java/updater/UpdateManager.java

```diff
@@ -90,6 +96,30 @@ private boolean initUpdate() {
             return false;
         }
 
+        return extractReplaceAndExecuteJar();
+    }
+
+    private boolean extractReplaceAndExecuteJar() {
+        logger.info("Extracting restarter JAR");
+        File replaceAndExecuteJar = new File(UPDATE_DIRECTORY + File.separator + UPDATE_RESTARTER_APP_NAME);
+
+        if (!replaceAndExecuteJar.exists()) {
+            try {
+                replaceAndExecuteJar.createNewFile();
+            } catch (IOException e) {
+                logger.error("Can't create replacer-and-executor JAR");
+                return false;
+            }
+        }
+
+        try (InputStream in = UpdateManager.class.getClassLoader().getResourceAsStream("updater/replaceAndExecute");
+             OutputStream out = new FileOutputStream(replaceAndExecuteJar)) {
+            IOUtils.copy(in, out);
+        } catch (IOException e) {
+            logger.error("Can't copy replacer-and-executor JAR", e);
+            return false;
+        }
```

Comment:
Same comment as before. 1. Method doing multiple things. 2. Consider using exceptions instead of boolean returns.